packages/ra-ui-materialui/src/field/ReferenceField.tsx

```diff
@@ -63,9 +63,10 @@ export const ReferenceField = <
 >(
     props: ReferenceFieldProps<RecordType, ReferenceRecordType>
 ) => {
-    const { source, emptyText, link = 'edit', ...rest } = props;
```

Comment:
You still pass down the rest props, which means you'll pass down teh `source`, that is explicitly excluded in NonEmptyReferenceField

Reply:
Added back